File: packages/csharp/src/V2/SyncSessionsClient.cs
using System;
using System.Threading;
using System.Threading.Tasks;
using Tesote.Sdk.Internal;
using Tesote.Sdk.Models;

namespace Tesote.Sdk.V2;

/// <summary>v2 sync-sessions resource — list and lookup per account.</summary>
public sealed class SyncSessionsClient
{
    private readonly Transport _transport;

    internal SyncSessionsClient(Transport transport)
    {
        _transport = transport;
    }

    /// <summary>List sync sessions for an account, ordered by creation desc.</summary>
    public async Task<SyncSessionListResponse> ListAsync(
        string accountId,
        int? limit = null,
        int? offset = null,
        string? status = null,
        CancellationToken ct = default)
    {
        ArgumentException.ThrowIfNullOrEmpty(accountId);
        var path = "/v2/accounts/" + Uri.EscapeDataString(accountId) + "/sync_sessions";
        var query = new QueryBuilder()
            .Add("limit", limit)
            .Add("offset", offset)
            .Add("status", status)
            .BuildOrNull();
        var opts = RequestOptions.Get(path);
        opts.Query = query;
        var node = await _transport.RequestAsync(opts, ct).ConfigureAwait(false);
        return Json.Deserialize<SyncSessionListResponse>(node);
    }

    /// <summary>Fetch a single sync session by id.</summary>
    public async Task<SyncSession> GetAsync(string accountId, string sessionId, CancellationToken ct = default)
    {
        ArgumentException.ThrowIfNullOrEmpty(accountId);
        ArgumentException.ThrowIfNullOrEmpty(sessionId);
        var path = "/v2/accounts/" + Uri.EscapeDataString(accountId)
            + "/sync_sessions/" + Uri.EscapeDataString(sessionId);
        var opts = RequestOptions.Get(path);
        var node = await _transport.RequestAsync(opts, ct).ConfigureAwait(false);
        return Json.Deserialize<SyncSession>(node);
    }
}

File: packages/java/src/main/java/com/tesote/sdk/v2/SyncSessionsClient.java
package com.tesote.sdk.v2;

import com.fasterxml.jackson.databind.JsonNode;
import com.tesote.sdk.Transport;
import com.tesote.sdk.internal.Json;
import com.tesote.sdk.internal.QueryParams;
import com.tesote.sdk.models.SyncSession;
import com.tesote.sdk.models.SyncSessionsPage;

import java.util.Objects;

/**
 * Read-only access to per-account sync sessions.
 */
public final class SyncSessionsClient {
    private final Transport transport;

    public SyncSessionsClient(Transport transport) { this.transport = transport; }

    public SyncSessionsPage list(String accountId) {
        return list(accountId, new ListParams());
    }

    public SyncSessionsPage list(String accountId, ListParams params) {
        Objects.requireNonNull(accountId, "accountId");
        Objects.requireNonNull(params, "params");
        Transport.Options opts = Transport.Options.get(
                "/v2/accounts/" + AccountsClient.encode(accountId) + "/sync_sessions")
                .query(QueryParams.of()
                        .put("limit", params.limit)
                        .put("offset", params.offset)
                        .put("status", params.status)
                        .build());
        JsonNode node = transport.request(opts);
        return Json.treeToValue(node, SyncSessionsPage.class);
    }

    public SyncSession get(String accountId, String sessionId) {
        Objects.requireNonNull(accountId, "accountId");
        Objects.requireNonNull(sessionId, "sessionId");
        Transport.Options opts = Transport.Options.get(
                "/v2/accounts/" + AccountsClient.encode(accountId)
                        + "/sync_sessions/" + AccountsClient.encode(sessionId));
        return Json.treeToValue(transport.request(opts), SyncSession.class);
    }

    public static final class ListParams {
        public Integer limit;
        public Integer offset;
        public String status;

        public ListParams limit(int v) { this.limit = v; return this; }
        public ListParams offset(int v) { this.offset = v; return this; }
        public ListParams status(String v) { this.status = v; return this; }
    }
}
